zebra-consensus/src/checkpoint/list.rs

```diff
@@ -64,7 +64,18 @@ impl CheckpointList {
             Err("checkpoint hashes must be unique")?;
         }
 
-        Ok(CheckpointList(checkpoints))
+        // Make sure all the hashes are valid. In Bitcoin, [0; 32] is the null
+        // hash. It is also used as the parent hash of genesis blocks.
+        if block_hashes.contains(&BlockHeaderHash([0; 32])) {
+            Err("the null hash is not a valid checkpoint hash")?;
```

Comment:
I think we should always aim to have our error message be descriptions of how the function was prevented from completing successfully, rather than just statements of the invariant that was violated.

```suggestion
            Err("checkpoint list contains invalid checkpoint hash: found null hash")?;
```